scanomatic/generics/abstract_model_factory.py

```diff
@@ -1181,26 +1174,26 @@ def serialize_structure(obj, structure):
             )
 
     @staticmethod
-    def serialize(obj, dtype):
+    def serialize(obj, dtype) -> Optional[str]:
         if obj is None:
             return None
 
         elif isinstance(dtype, type) and issubclass(dtype, Enum):
-            return obj.name
+            return str(obj.name)
```

Comment:
I think this is fine.

Reply:
kan tycka det känns lite märkligt att metoden tar dtype som input parameter, är det nödvändigt, eller det kanske är så att dtype inte behöver överensstämma med type av obj, vilket rad 1195 verkar antyda? Hade ju varit tydligare om man  kunde typehinta obj till Optional[Union[vad det nu kan vara]]

Reply:
Det är många saker i den här delen av koden som känns som att de kanske är överdesignade, men jag vågar mig nog inte på några större refaktoreringar innan jag tror att det funkar, vilket jag är rätt säker på att det integör, även om unit-testerna går igenom (integrationstesterna visar det, tycker jag).

Reply:
jag underar seriöst över detta det verkar ju inte vara en instans av en enum utan en typ och då finns ju inte `obj.name`

```python
In [1]: from enum import Enum

In [2]: class C(Enum):
   ...:     D = 1
   ...: 

In [3]: C.name
----------------------------------------------------------------
AttributeError                 Traceback (most recent call last)
<ipython-input-3-bde6c5522d40> in <module>
----> 1 C.name

/usr/lib/python3.9/enum.py in __getattr__(cls, name)
    403             return cls._member_map_[name]
    404         except KeyError:
--> 405             raise AttributeError(name) from None
    406 
    407     def __getitem__(cls, name):

AttributeError: name

In [4]: isinstance(C, type)
Out[4]: True

In [5]: isinstance(C.D, type)
Out[5]: False
```

Reply:
I think I understand what it does now. What a convoluted thing. Who wrote this 😎 
Maybe replace that if-block with
```python
if isinstance(obj, Enum):
    return obj.name
```
That should be properly typed and all.

Reply:
Sure, if you think that's safe.